grafana/oncall
# Grafana Faro integration

# What this PR does

- Added Faro integration
- Added support for `.env` files on frontend

# How to test

- Set up `grafana-agent` to pass data to `Loki`, `Tempo` and `Prometheus`
- Create your `.env` file in `grafana-plugin` - pass in `FARO_URL` and `FARO_API_KEY` as defined in grafana-agent config
 
```
FARO_URL=http://localhost:12345/collect
FARO_API_KEY=secret
FARO_ENABLED=true
```
- Run `grafana-agent`
- Run `yarn watch`
- Go to grafana instance and see logs/traces in the matching data sources

Example of grafana agent config file

```
logs:
  configs:
  - clients:
    - basic_auth:
        password: 
        username: 343538
      url: https://logs-prod-017.grafana.net/loki/api/v1/push
    name: integrations
    positions:
      filename: /tmp/positions.yaml
    scrape_configs:
    - job_name: integrations/agent
      journal:
        labels:
          instance: <hostname>
          job: integrations/agent
        max_age: 24h
      pipeline_stages:
      - match:
          action: drop
          drop_counter_reason: only_keep_grafana_agent_logs
          selector: '{unit!="grafana-agent.service"}'
      - regex:
          expression: (level=(?P<log_level>[\s]*debug|warn|info|error))
      - labels:
          level: log_level
      relabel_configs:
      - source_labels:
        - __journal__systemd_unit
        target_label: unit
    - job_name: integrations/agent
      pipeline_stages:
      - json:
          expressions:
            message: message
      - regex:
          expression: (level=(?P<log_level>[\s]*debug|warn|info|error))
      - labels:
          level: log_level
      - output:
          source: message
      relabel_configs:
      - source_labels:
        - computer
        target_label: instance
      - replacement: grafana-agent.service
        target_label: unit
      windows_events:
        bookmark_path: ./bookmark-application.xml
        eventlog_name: Application
        labels:
          job: integrations/agent
        use_incoming_timestamp: true
        xpath_query: '*[System[Provider[@Name=''Grafana Agent'']]]'
    target_config:
      sync_period: 10s
metrics:
  configs:
  - name: integrations
    remote_write:
    - basic_auth:
        password: 
        username: 689135
      url: https://prometheus-us-central1.grafana.net/api/prom/push
  global:
    scrape_interval: 60s
  wal_directory: /tmp/grafana-agent-wal
traces:
  configs:
    - name: integrations
      remote_write:
        - endpoint: tempo-us-central1.grafana.net:443
          basic_auth:
            username: 340051
            password: 
      receivers:
        otlp:
          protocols:
            grpc:
            http:
              cors:
                allowed_origins:
                  - http://localhost:12345
                  - http://localhost:3000
                max_age: 7200
integrations:
  app_agent_receiver:
      autoscrape:
        enable: true
        metrics_instance: integrations
      instance: 'frontend'
      logs_instance: integrations
      traces_instance: integrations
      server:
        host: 0.0.0.0
        port: 12345
        api_key: 'secret'
        cors_allowed_origins:
          - http://localhost:3000
      logs_labels:
        app: frontend
        kind:
      logs_send_timeout: 5000
      sourcemaps:
        download: true
```
